examples/chip-tool/commands/common/Commands.cpp

```diff
@@ -119,12 +119,13 @@ static void DetectAndLogMismatchedDoubleQuotes(int argc, char ** argv)
 
 } // namespace
 
-void Commands::Register(const char * clusterName, commands_list commandsList, const char * helpText)
+void Commands::Register(const char * clusterName, commands_list commandsList, const char * helpText, bool isSynthetic)
```

Comment:
Why not reverse it and make it "isCluster" instead of "isSynthetic". If anything the fact that cluster names are top-level commands is more synthetic than classic "there are commands on this command line".

Reply:
Yep, I can do that.  This started off with a slightly different shape before I ended up with the toplevel "command set" and "cluster" APIs.  ;)